xdsl/dialects/dlti.py

```diff
@@ -30,6 +37,23 @@ class DataLayoutEntryAttr(ParametrizedAttribute):
     key: Attribute
     value: Attribute
```

Comment:
This we could constrain here directly
```suggestion
    value: StringAttr | IntegerAttr | FloatAttr[Float32Type]
```

Reply:
The value can be anything and is not just strings, integers or floats (e.g. it can be types, any of the attributes of this operation etc).

Reply:
Yer, that's a very good point and likely to cause some confusion. Following this, as suggested, I've updated to remove creation of FloatAttr from float, and now when passed int it creates IntAttr instead of IntegerAttr.

Reply:
Nice I think the new version should avoid confusion